staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -814,10 +821,16 @@ func ValidateCustomResourceDefinitionOpenAPISchema(schema *apiextensions.JSONSch
 	}
 
 	if schema.Items != nil {
-		allErrs = append(allErrs, ValidateCustomResourceDefinitionOpenAPISchema(schema.Items.Schema, fldPath.Child("items"), ssv, false, opts)...)
+		subSsv := ssv
+
+		if schema.XListType == nil || *schema.XListType != "map" {
```

Comment:
a comment here would be helpful:
```
// we can only correlate old/new items for "map" and "set" lists (unset list type defaults to "atomic")
```

Reply:
also, restricting to `map` makes sense for CEL expressions (since correlating items by index in `atomic` lists doesn't make a ton of sense for cel, and correlated items in a `set` list would always have to be identical)

however, correlated list items to populate `oldSelf` for internal ratcheting might still make sense to do by index for `atomic` lists and by identity for `set` lists

just want to make sure we don't pin logic of `oldSelf` calculation in code to eligibility to use `oldSelf` in cel expressions